n2survey/lime/survey.py

```diff
@@ -531,6 +609,104 @@ def plot(
+        """
+        function to check if question type and/or combination of questions for
+        compare_with is already implemented and working
+        """
+        if not add_questions:
+            add_questions = []
+        supported_plots = ["single-choice", "multiple-choice", "array"]
+        supported_comparisons = [("single-choice", "single-choice")]
+        all_plots = [question]
+        if compare_with:
+            all_plots.append(compare_with)
+        for i in add_questions:
+            all_plots.append(i)
+        for plot in all_plots:
+            if self.get_question_type(plot) not in supported_plots:
+                raise NotImplementedError(
+                    """
+                    Question type not yet implemented
+                    """
+                )
+        if compare_with:
+            tuple_list = [(question, compare_with)]
+            for i in add_questions:
+                tuple_list.append((question, i))
+            for i in tuple_list:
+                question_type_tuple = (
+                    self.get_question_type(i[0]),
+                    self.get_question_type(i[1]),
+                )
+                if question_type_tuple not in supported_comparisons:
+                    raise NotImplementedError(f" comparison {i} not yet implemented")
+
+    def get_answer_sequence(self, question, add_questions=None, totalbar=False):
+        """
+        creates answer sequence from given questions to keep sequence in
+        plot consistent if answers that were never chosen are suppressed
+        """
+        if not add_questions:
+            add_questions = []
+        answer_sequence = [
+            list(self.count(question, labels=True).index.values.astype(str))
+        ]
+        if add_questions:
+            for entry in add_questions:
+                answer_sequence.append(
+                    list(self.count(entry, labels=True).index.values.astype(str))
+                )
+        if totalbar:
+            answer_sequence[0].insert(0, "Total")
+        return answer_sequence
+
+    def create_comparison_data(
+        self, question, compare_with, answer_sequence, add_questions=None
+    ):
+        """
+        This function loads and combines necessary data for the plot functions.
+        depending on the wanted (question/add_question_entry,compare_with) tuple
+        """
+        if not add_questions:
+            add_questions = []
+        df = []
+        if self.get_question_type(compare_with) == "single-choice":
+            # create Dataarray from all existing combinations of
+            # question and compare_with
+            df.append(
+                pd.concat(
+                    [
+                        self.get_responses(question, labels=True, drop_other=True),
+                        self.get_responses(compare_with, labels=True, drop_other=True),
+                    ],
+                    axis=1,
+                ).values
```

Comment:
It feels rather unnatural to turn Dataframes returned by `get_responses` into ndarrays because most of what you do with them downstream work better with DF in general, e.g. slicing in Line 688 and 706 as well as Line 17 in `simple_comparison.py` below. I know you said you are more comfortable with numpy than pandas, so perhaps @rguliev can decide if this is ok.

Reply:
I tried to work everything with dataframes, and it clearly did not work better. So after a lot of hours trying to add another set of commands to my memory (the Pandas one), 
I chose to keep it the way I handled plotting all the time before, which would be the use of arrays and lists as the easiest structure. This keeps the number of necessary libraries as small as possible. For my Data evaluation numpy is needed all the time but pandas, as soon as translatet to arrays is absolutely not important here at all. 
This makes scripts also easier for new programmers that only have to learn to work with numpy and matplotlib.pyplot and not additionally with pandas.
I admit this might be a question of personal preferences, but the main goal for me was a working script as soon as possible and Pandas gobbled away a lot of time so I dropped it as soon as possible. I have no problem if somebody afterwards turns every action I did with numpy to pandas if it is more convenient for them, but I will not work with that library more then I have to.